### ISSUE TITLE
http_conn is not really private

### ISSUE THREAD

**Issue by gquintard** [varnishcache/varnish-cache#3802]
TL;DR: in `master`, even though `http_conn` is private, backends are expected to know about it.

[http_conn](https://github.com/varnishcache/varnish-cache/blob/master/bin/varnishd/cache/cache_varnishd.h#L95) is located in a file saying [Stuff that should *never* be exposed to a VMOD](https://github.com/varnishcache/varnish-cache/blob/master/bin/varnishd/cache/cache_varnishd.h#L31), however, both [cache_fetch.c](https://github.com/varnishcache/varnish-cache/blob/master/bin/varnishd/cache/cache_fetch.c#L444) and [cache_vrt_filter.c](https://github.com/varnishcache/varnish-cache/blob/master/bin/varnishd/cache/cache_vrt_filter.c#L343) use it (and don't even check it exists before accessing it)

This is problematic notably for vmods as they need to create a dummy `http_conn` that they should not know about.

**Comment by Dridi** [comment]
FWIW `cache_varnishd.h` says that, but it is also installed as a header so you can use it from a VMOD. One could say you may manage your htc but here be dragons, for now. As you pointed out, a VMOD implementing its own backend transport needs a clear http_conn contract before it can move to `cache.h` or some other public header for VMODs.

**Comment by gquintard** [comment]
in the same vein: `vcldiir->health_changed` is a bit murky too. It's obviously there, and we can modify it with `VRT_SetChanged`, but we can't retrieve it's value, which would be useful when implementing the `healthy` method
